cmake/OpenCVCompilerOptions.cmake

```diff
@@ -110,9 +110,15 @@ elseif(CV_GCC OR CV_CLANG)
     add_extra_compiler_option(-ffast-math)
     add_extra_compiler_option(-fno-finite-math-only)
   endif()
+elseif(CV_ICX)
+  # ICX uses -ffast-math by default.
+  if(NOT ENABLE_FAST_MATH)
+    add_extra_compiler_option(-fno-fast-math)
+    add_extra_compiler_option(-fp-model=precise)
```

Comment:
It makes sense to guard this similar to line 101 (like CV_ICC) and/or 93 (MSVS)

Reply:
Done. Please take a look.